pkg/ccl/changefeedccl/enriched_source_provider.go

```diff
@@ -163,9 +150,115 @@ func (p *enrichedSourceProvider) GetJSON(updated cdcevent.Row) (json.JSON, error
 func (p *enrichedSourceProvider) GetAvro(
 	row cdcevent.Row, schemaPrefix string,
 ) (*avro.FunctionalRecord, error) {
-	sourceDataSchema, err := avro.NewFunctionalRecord("source", schemaPrefix, p.getAvroFields(), p.avroSourceFunction)
+	sourceDataSchema, err := avro.NewFunctionalRecord("source", schemaPrefix, avroFields, p.avroSourceFunction)
 	if err != nil {
 		return nil, err
 	}
 	return sourceDataSchema, nil
 }
+
+type fieldInfo struct {
+	avroSchemaField    avro.SchemaField
+	kafkaConnectSchema kcjsonschema.Schema
+}
+
+var allFieldInfo = map[string]fieldInfo{
+	"job_id": {
```

Comment:
nit: maybe these field names should be constants, mostly to provide peace of mind that we're using the same name in the avro and kafka connect versions of the schema?